modules/light-clients/06-solomachine/module.go

```diff
@@ -1,6 +1,53 @@
 package solomachine
 
-// Name returns the solo machine client name.
-func Name() string {
+import (
+	"encoding/json"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/spf13/cobra"
+)
+
+var _ module.AppModuleBasic = AppModuleBasic{}
+
+// AppModuleBasic defines the basic application module used by the solo machine light client.
+type AppModuleBasic struct{}
+
+// Name returns the solo machine module name.
+func (AppModuleBasic) Name() string {
 	return SubModuleName
 }
+
+// RegisterLegacyAminoCodec does nothing. The solo machine client does not support amino.
+func (AppModuleBasic) RegisterLegacyAminoCodec(*codec.LegacyAmino) {}
+
+// RegisterInterfaces registers module concrete types into protobuf Any.
+func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	RegisterInterfaces(registry)
+}
```

Comment:
is it worth mentioning in the other docstrings, or maybe in the `AppModuleBasic` docstring that this is the only method that needs to be implemented? I don't think docstrings like `DefaultGenesis returns nil` provide much value and would rather know why all these functions are no-oping rather than providing an implementation. WDYT?

Reply:
Good suggestion! Checkout this commit https://github.com/cosmos/ibc-go/pull/2826/commits/45e6bb534d170372cde80cc67090fc05ef0c93f2, let me know if there are any improvements I could add